peer_review_data/main.py

```diff
@@ -1,41 +1,201 @@
+                        f'({canvasAssessment.id}).')
+            continue
+
+        assessment: models.Assessment = None
+        try:
+            assessment = \
+                models.Assessment.fromCanvasAssessment(canvasAssessment)
+            LOGGER.debug(f'Saving {assessment}…')
+            assessment.save()
+        except Exception as e:
+            # XXX: Catches assessments referring to non-existent submissions!
+            LOGGER.warning(f'Error saving Assessment '
+                           f'({canvasAssessment.id}): {e}')
+            LOGGER.debug('Assessment data: ' + json.dumps(
+                dictSkipKeys(canvasAssessment, ['_requester']),
+                indent=2, default=str))
+
+        if assessment:
+            canvasComment: CanvasComment
+            for canvasComment in [CanvasComment(c) for c in
+                                  canvasAssessment.comments]:
+                try:
+                    comment: models.Comment = \
+                        models.Comment.fromCanvasCommentAndAssessment(
+                            canvasComment, assessment)
+                    LOGGER.debug(f'Saving {comment}…')
+                    comment.save()
+                except Exception as e:
+                    LOGGER.warning(f'Error saving {comment}: {e}')
+
+
 def main() -> None:
     timeStart: datetime = datetime.now(tz=utc)
     LOGGER.info(f'Start time: {timeStart.isoformat(timespec="milliseconds")}')
 
-    course: CanvasCourse = canvas.get_course(COURSE_ID)
-    LOGGER.info(f'Found course ({course.id}): "{course.name}"')
+    canvasCourse: CanvasCourse = canvas.get_course(COURSE_ID)
+    LOGGER.info(f'Found course ({canvasCourse.id}): "{canvasCourse.name}"')
+    courseSaved = False
+    course: models.Course | None = None
+    user: models.User | None = None
 
-    assignment: CanvasAssignment = course.get_assignment(ASSIGNMENT_ID)
-    LOGGER.info(f'Found assignment ({assignment.id}): "{assignment.name}"')
+    canvasAssignment: CanvasAssignment = canvasCourse.get_assignment(
+        ASSIGNMENT_ID)
+    LOGGER.info(
+        f'Found assignment ({canvasAssignment.id}): "{canvasAssignment.name}"')
 
-    if assignment.peer_reviews is not True:
+    if canvasAssignment.peer_reviews is not True:
         LOGGER.info(
             f'Skipping assignment ({ASSIGNMENT_ID}) in course ({COURSE_ID}): '
             'Not configured for peer reviews.')
         sys.exit()
 
-    LOGGER.info(f'Assignment ({assignment.id}) is peer reviewed')
+    LOGGER.info(f'Assignment ({canvasAssignment.id}) is peer reviewed')
+
+    assignmentRubricId: int = canvasAssignment.rubric_settings.get('id')
+    LOGGER.info(f'Assignment ({canvasAssignment.id}) has '
+                f'rubric ID ({assignmentRubricId})')
+
+    canvasAssignmentRubric: CanvasRubric = canvasCourse.get_rubric(
+        assignmentRubricId, include='assessments', style='full')
+
+    if not hasattr(canvasAssignmentRubric, 'assessments'):
```

Comment:
what the difference between line 164 and line 173?

Reply:
Good catch.  That's a mistake.  The second if-statement should be using a different condition to handle those cases where the assignment has been configured for peer reviews, but there aren't any reviews yet.

Reply:
@zqian, I've updated the file.  Please resolve this conversation if you think this has been handled satisfactorily.

Reply:
Looks great!